src/schema/schema-generator.ts

```diff
@@ -255,12 +268,17 @@ export abstract class SchemaGenerator {
         return superClassTypeInfo ? superClassTypeInfo.type : undefined;
       };
       const interfaceClasses = objectType.interfaceClasses || [];
+
       return {
         target: objectType.target,
         isAbstract: objectType.isAbstract || false,
         type: new GraphQLObjectType({
           name: objectType.name,
           description: objectType.description,
+          astNode: this.createClassDirectiveNodes(
+            objectType.name,
+            objectType.directives,
+          ) as ObjectTypeDefinitionNode,
```

Comment:
Why we have to make `as` assertions for every astNode?

Reply:
Removing it gave me a type error due to the directives needing to be ObjectTypeDefinitionNode or InputObjectTypeDefinitionNode.

Reply:
Would it be too hard to change the return type of the `createFieldDirectiveNodes` function? It can be piped if backwards compatibility is a concern.

Reply:
I think we can remove some assertions by using proper overloads. Maybe a generic would looks better than assertion where it has to be?

Reply:
@19majkel94 @ldiego08 I ended up creating a new method for each type and removed the whole reverse() then concatenating thing.  So `@Directive()` only supports one directive per annotation which I think is cleaner anyway.